### DISCUSSION TITLE
How do I get this running on local?

### DISCUSSION THREAD

**Discussion by openSourceBugs** [fastapi/full-stack-fastapi-template#1110] [Questions] [answered]
I added configuration to the .env files and ran docker compose up -d
I see the login page at localhost but I also see 
POST https://localhost/api/v1/login/access-token net::ERR_CONNECTION_REFUSED

I'm not sure if it has to do with the FLOWER_BASIC_AUTH= in .env

There's not much to go by in the docs.  It says to not use this and use it at the same time. 

"This project is currently being restructured, don't use it right now, hold for a bit.

In the next couple of weeks it will be ready. 😎 🚀"

"You can just fork or clone this repository and use it as is.

✨ It just works. ✨"

I would also like to debug in the backend at least.

**Comment by openSourceBugs** [comment]
# Update this with your app domain
DOMAIN=localhost
# DOMAIN=localhost.tiangolo.com

SERVER_HOST=http://localhost

PROJECT_NAME="FastAPI Project"

STACK_NAME=fastapi-project

# Backend
BACKEND_CORS_ORIGINS="http://localhost,http://localhost:5173,https://localhost,https://localhost:5173"
SECRET_KEY=foo
FIRST_SUPERUSER=admin@example.com
FIRST_SUPERUSER_PASSWORD=foo
SMTP_HOST=
SMTP_USER=
SMTP_PASSWORD=
EMAILS_FROM_EMAIL=info@example.com
SMTP_TLS=True
SMTP_PORT=587

USERS_OPEN_REGISTRATION=False

# Postgres
POSTGRES_SERVER=db
POSTGRES_USER=postgres
POSTGRES_DB=app
POSTGRES_PASSWORD=foo

# PgAdmin
PGADMIN_DEFAULT_EMAIL=admin@example.com
PGADMIN_DEFAULT_PASSWORD=foo
PGADMIN_LISTEN_PORT=5050

SENTRY_DSN=

# Flower
FLOWER_BASIC_AUTH=

# Configure these with your own Docker registry images
DOCKER_IMAGE_BACKEND=backend
DOCKER_IMAGE_CELERYWORKER=celery
DOCKER_IMAGE_FRONTEND=frontend

**Comment by openSourceBugs** [comment]
Probably going to give up on this project for a bit since the documentation isn't working for me.  Too bad. I was excited to see a project that takes away a lot of boilerplate.

**Reply by chris-dare** [reply]
I don't think you needed to wait. The project is version controlled so if the current version is unstable, you can always clone a previously stable release, like [0.5.0](https://github.com/tiangolo/full-stack-fastapi-template/releases/tag/0.5.0)

But v0.6.0 looks good now, based on the documentation and release notes. Take it out for a spin and see what you like. Cheers mate.

**Reply by openSourceBugs** [reply]
The problem is still mixing http with https and somehow some configuration somewhere, which no one seems to know about is causing this issue.  The original ticket pretty much described what the problem was (https).

**Comment by opyate** [comment, marked as answer]
Check the docker-compose logs. Did the backend start up without errors?

You're looking for a clean startup:

```
backend_1   | INFO:     Will watch for changes in these directories: ['/app']
backend_1   | INFO:     Uvicorn running on http://0.0.0.0:80 (Press CTRL+C to quit)
backend_1   | INFO:     Started reloader process [1] using WatchFiles
backend_1   | INFO:     Started server process [11]
backend_1   | INFO:     Waiting for application startup.
backend_1   | INFO:     Application startup complete.
```

This has to work from your browser: http://localhost/api/v1/openapi.json

![openapi](https://github.com/tiangolo/full-stack-fastapi-template/assets/96808/634cc29d-eaf0-4d45-8e9c-68fdfbac54a9)


If you get it working, let us know how you did it, then close the issue.

**Reply by tiangolo** [reply]
Yep, as @opyate says, please check the Docker Compose logs, the backend logs, the frontend console logs, etc. Please share more info and that we we could help you.

**Reply by openSourceBugs** [reply]
Uvicorn running on http://0.0.0.0:80 (Press CTRL+C to quit)    Why is the front end requesting https for login? Is there some special setup for this?

**Reply by JacobFV** [reply]
ubuntu 22.04 here. copied using copier. works on dev (haven't tested prod) no changes other than filling in parameters

**Comment by estvita** [comment]
Hi, I have a similar problem
and I see this in the logs
```
backend-1   | INFO:     Will watch for changes in these directories: ['/app']
backend-1   | INFO:     Uvicorn running on http://0.0.0.0:80 (Press CTRL+C to quit)
backend-1   | INFO:     Started reloader process [1] using WatchFiles
backend-1   | ERROR:    Error loading ASGI app. Import string ":app" must be in format "<module>:<attribute>".
```

**Comment by chupa-cabras** [comment]
I have exactly same problem:
POST https://localhost/api/v1/login/access-token net::ERR_CONNECTION_REFUSED

In front end they are render this URL httpS://localhost/ with HTTPS. 

No errors in docker compose, but the backend cannot be able to respond in 443 . I ll try to configure that . But if you have a tip it ll be helpfully .

**Reply by chupa-cabras** [reply]
WORKS !!!!!!
I delete all containers e rebuild. It works well. \o/ 
TKSSSSSS

**Comment by openSourceBugs** [comment]
The problem is still that the servers are not handling https requests.  I don't get how this has been an issue for so long.  The swagger open api json url works.  It's a https issue.  There are no real errors in the back end logs.  
![image](https://github.com/tiangolo/full-stack-fastapi-template/assets/5646010/0cd2e6ec-21c8-4f13-88f0-529ab0783eb4)
There are new errors in the front end though and I see that the flower config was removed from the .env file.  This project is way too unstable.

**Comment by JakeNYC** [comment]
Hey everyone. Had the same issue above.  (POST https://localhost/api/v1/login/access-token net::ERR_CONNECTION_REFUSED)

Here are the steps to fix, not sure if it was clear in the above.  Hopefully someone will fix in repo (I'm not 100% how to propose this offically)

Step1
1. Update the docker-compose.override.yml file, changing the frontend build args...(just removing the "s" from https)
     From: VITE_API_URL=https://api.${DOMAIN?Variable not set}
     To:  VITE_API_URL=http://api.${DOMAIN?Variable not set}

2.   Rebuild the containers (for some reason this was needed on my system, and looks like was true for others)
     # Stop and remove containers
     docker compose down

     # Remove all images (optional) 
     docker compose down --rmi all

     # Rebuild and start
     docker compose up -d --build

3.  Run:  docker compose watch
